### PR TITLE
Fold mapper setters only for the variable the mapper directly initializes

### PR DESCRIPTION
**Pull request by MBoegers**
## What's changed?

`MigrateMapperSettersToBuilder` resolved the mapper's owning variable with an unconstrained `firstEnclosing(NamedVariable/Assignment)`. The owner is now resolved by walking up from the mapper expression through value-preserving wrappers only (parentheses, casts, Kotlin statement adapters, chained identifier assignments), and the relocation anchor for intermediate statements is derived from that same walk. Guards were added on the same collection path: setters inside control flow or consumed by a declaration initializer stay put (rewritten as `rebuild()` in place or annotated with a TODO), void-returning methods only count as unknown setters when named `set*`, rebuild reassignments only fold into initializers that can act as a method-call select without parentheses, the TODO comment fallback breaks the line when embedded in a larger statement, and Kotlin `val` locals are never un-finalized.

## What's your motivation?

A `new JsonMapper()` nested inside another expression — a constructor argument, ternary branch, or array initializer — made the recipe claim the first enclosing variable as the mapper's owner and fold that variable's own setters into the builder chain, producing uncompilable output:

```java
JobCredentialListService service = new JobCredentialListService(new JsonMapper());
service.setApplicationEventPublisher(spy);   // was folded into JsonMapper.builder() with TODOs
```

It could also silently rewrite unrelated configuration (`service.setDateFormat(df)` became `.defaultDateFormat(df)` on the mapper builder), and `if (production) { mapper.enable(...); }` was hoisted unconditionally into the builder.

- Fixes https://github.com/moderneinc/customer-requests/issues/2750

## Anything in particular you'd like reviewers to focus on?

- `enclosingOwnerCursor` / `isTransparentWrapper`: the definition of "transparent" (parens, `J.TypeCast`, Kotlin `ExpressionStatement`/`StatementExpression`, chained identifier assignments) decides which owners still fold.
- The statement-level gate in `collectStandaloneSetters` (`isSimpleExpressionStatement`) narrows what qualifies as a standalone setter; conditional/consumed setters now go through the rebuild/TODO fallbacks instead.

## Any additional context

Known pre-existing gaps left out of scope (confirmed while reviewing, happy to file separately): setter collection on a shadowing same-name variable in an anonymous class body, `rebuild()` typing for user subclasses of format-aligned mappers, relocation ordering for multi-variable declarations, and TODO placement for setters in single-line blocks / Kotlin `return mapper.enable(x)` expressions.

## Checklist

- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the [IntelliJ IDEA auto-formatter](https://docs.openrewrite.org/authoring-recipes/recipe-development-environment#use-the-intellij-idea-auto-formatter) on affected files

### CODE REVIEW

File: src/test/java/org/openrewrite/java/jackson/MigrateMapperSettersToBuilderTest.java [hunk 1]
```diff
@@ -337,6 +337,108 @@ JsonMapper create(Module module) {
...
+                                  .disable(SerializationFeature.INDENT_OUTPUT)
+                                  .build();
+                      }
+
+                      JsonMapper viaChainedAssignment() {
+                          JsonMapper a;
+                          JsonMapper b = a = JsonMapper.builder()
+                                  .enable(SerializationFeature.INDENT_OUTPUT)
+                                  .build();
+                          return a;
```
Comment (steve-aom-elliott) [inline src/test/java/org/openrewrite/java/jackson/MigrateMapperSettersToBuilderTest.java:388]: Is this potentially a semantic change given the chaining is present during the assignment to `a`? Trying to remember whether that's still by reference by that point in the original.

Comment (MBoegers) [reply src/test/java/org/openrewrite/java/jackson/MigrateMapperSettersToBuilderTest.java:388]: Interesting catch; this edge case uncovered a new edge case...

Comment (steve-aom-elliott) [reply src/test/java/org/openrewrite/java/jackson/MigrateMapperSettersToBuilderTest.java:388]: Was the change for this going to be part of this PR or something later, or did you figure out that it's somehow not a semantic change?

Comment (MBoegers) [reply src/test/java/org/openrewrite/java/jackson/MigrateMapperSettersToBuilderTest.java:388]: Both, in the end. For the shape in this test it's not a semantic change, `a` and `b` alias the _same instance_ and nothing observes it between construction and the setter, so folding into the shared initializer is equivalent. 
But your question uncovered the case where it would be one: a read through the alias between the initializer and the setter. That's guarded in [ddc89c0](https://github.com/openrewrite/rewrite-jackson/pull/157/changes/ddc89c033cfa002e8ae56383377b6fce61edd831) (last commit this PR): the owner search now reports chained-assignment aliases, and setter collection stops at the first statement referencing one, see `doNotFoldOverReferencesToChainedAssignmentAlias`.
